src/modules/flow/http-client/http-client.c

```diff
@@ -377,20 +355,11 @@ common_request_finished(void *data,
 
     if (streq(response->content_type, "application/json") &&
         type->process_token) {
-        struct sol_json_scanner scanner;
-        struct sol_json_token token, key, value;
-        enum sol_json_loop_reason reason;
-
-        sol_json_scanner_init(&scanner, response->content.data, response->content.used);
-        SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
-            if (!sol_json_token_str_eq(&token, mdata->key.data, mdata->key.len))
-                continue;
-            ret = type->process_token(node, &key, &value);
-            if (ret < 0)
-                goto err;
-        }
+        ret = type->process_token(node, &response->content);
```

Comment:
seems you only need the slice (ptr/size), so you can change the signature to take that, then you use `sol_buffer_get_slice(&response->content)`